src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] buckets;
+
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private double threshold = capacity * LOAD_FACTOR;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (buckets[0] == null) {
+                buckets[0] = new Node<>(null, value);
+            } else {
+                Node<K, V> current = buckets[0];
+                while (current != null) {
+                    if (current.key == null) {
+                        current.value = value;
+                        return;
+                    }
+                    if (current.next == null) {
+                        break;
+                    }
+                    current = current.next;
+                }
+                current.next = new Node<>(null, value);
+            }
+            size++;
 
+            if ((double) size / buckets.length > LOAD_FACTOR) {
+                resize();
+            }
+            return;
+        }
+
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (buckets[index] == null) {
+            buckets[index] = newNode;
+        } else {
+            Node<K, V> current = buckets[index];
+            while (current != null) {
+                if (current.key != null && current.key.equals(key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    break;
+                }
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+        if (size >= threshold) {
+            resize();
+        }
+
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = buckets[index];
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                return current.value;
+            }
+            if (current.key == null && key == null) {
+                return current.value;
+            }
```

Comment:
This could be a single condition - use Objects.equals() for null-safe comparison.